studio/components/LanguageSelector.tsx

```diff
@@ -0,0 +1,151 @@
+import { Box, Button, Card, Checkbox, Flex, Text, useTheme } from "@sanity/ui";
+import React from "react";
+import { PatchEvent, set } from "sanity";
+
+import {
+  Language,
+  supportedLanguages,
+} from "internationalization/supportedLanguages";
+
+interface LanguageSelectorProps {
+  value?: Language[];
+  onChange: (event: PatchEvent) => void;
+}
+
+const colorMap = {
+  dark: {
+    selected: "#29356a",
+    default: "rgba(19, 20, 27, 0.5)",
+  },
+  light: {
+    selected: "#E3F2FD",
+    default: "white",
+  },
+};
+
+const LanguageSelector = ({ value = [], onChange }: LanguageSelectorProps) => {
+  const theme = useTheme();
+  const prefersDark = theme.sanity.v2?.color._dark ?? false;
+
+  // Get the currently set default language
+  const currentDefaultLanguage = value.find((lang) => lang.default)?.id || null;
+
+  const handleLanguageSelection = (lang: Language) => {
+    const isSelected = value.some((item) => item.id === lang.id);
+    // Prevent deselecting the last remaining language
+    if (isSelected && value.length === 1) {
+      console.log("Cannot deselect the last remaining language.");
+      return; // Exit early if there's only one language selected
+    }
+
+    const updatedValue = isSelected
+      ? value.filter((item) => item.id !== lang.id) // Deselect language
+      : [...value, { ...lang, default: false }]; // Select language
+
+    const newDefaultLanguage = getNewDefaultLanguage(
+      updatedValue,
+      currentDefaultLanguage,
+      lang,
+    );
+
+    const finalValue = updatedValue.map((item) => ({
+      ...item,
+      default: item.id === newDefaultLanguage,
+    }));
+
+    onChange(PatchEvent.from(set(finalValue)));
```

Comment:
```suggestion
    if (isSelected && lang.id === currentDefaultLanguage) {
      // default was deselected, set first language as default
      updatedValue[0].default = true;
    }

    onChange(PatchEvent.from(set(updatedValue)));
```

Reply:
The original doesn't set the first language as the default but the next selected, so if 2 and 3 is selected where 2 is default, and I unselect 2, number 3 is set as default.

Reply:
Is that important? Does the user expect this behaviour, or would they equally accept the new default as the first language.

Reply:
I don't the original has that behaviour either. Select 1,2 and 3, mark 2 as default and deselect 2. You would then expect 3 to be set as default, but it's actually 1.

Reply:
The fact that the languages are not necessarily in the same order in the selection list as in `supportedLanguages` makes it a little hard to reason about this behaviour. I think it would be clearer if we look at only the IDs when selecting and deselecting, and then rebuild the selection array by filtering `supportedLanguages` with the selected IDs to preserve the same order as `supportedLanguages`.

Reply:
Yes, that's correct, it takes the first selected object in the array. We could ofc take the next in line, but I don't know if that's a good experience.  I gave it a try and found it a little confusing at the time

And we do want that UX for cases where you've chosen 2 and 3, but deselect 2, you don't want 1 to pop up as selected with default.

Reply:
The fact that the languages are not necessarily in the same order in the selection list as in supportedLanguages makes it a little hard to reason about this behavior. I think it would be clearer if we look at only the IDs when selecting and deselecting, and then rebuild the selection array by filtering supportedLanguages with the selected IDs to preserve the same order as supportedLanguages.

What about something like this?

```ts
const handleLanguageSelection = (lang: Language) => {
  const selectedIds = value?.map((language) => language.id);
  const isSelected = selectedIds?.includes(lang.id);
  // Prevent deselecting the last remaining language
  if (isSelected && selectedIds.length === 1) {
    return; // Exit early if there's only one language selected
  }
  const updatedSelectedIds = isSelected
    ? selectedIds.filter((id) => id !== lang.id) // Deselect language
    : [...selectedIds, lang.id]; // Select language
  const needNewDefault =
    (isSelected && lang.id === currentDefaultLanguage) ||
    currentDefaultLanguage === null;
  const updatedValue = supportedLanguages.reduce<Language[]>(
    (acc, language) => {
      if (!updatedSelectedIds.includes(language.id)) {
        return acc;
      }
      return [
        ...acc,
        {
          ...language,
          default: needNewDefault
            ? acc.length === 0
            : currentDefaultLanguage === language.id,
        },
      ];
    },
    [],
  );
  onChange(PatchEvent.from(set(updatedValue)));
};
```

Reply:
Why do you want value (selected languages) to be the same order as supportedLanguages?

Reply:
So that the new default does not depend on the order in which you selected the languages.

Reply:
Ah, gotcha!